spec/rex/parser/arguments_spec.rb

```diff
@@ -50,14 +52,72 @@
 
       it 'ignores combined flags that do not exist' do
         input = ['-unknown-flags']
-        # Not sure if this flag dropping is intentional behavior
         expected_yields = [
-          ['-o', 0, nil],
+          [nil, 0, '-unknown-flags']
         ]
         expect { |b| subject.parse(input, &b) }.to yield_successive_args(*expected_yields)
       end
     end
 
+    it 'parses a single long flag correctly' do
+      input = ['--help']
+      expected_yields = [
+        ['--help', 0, nil]
+      ]
+      expect { |b| subject.parse(input, &b) }.to yield_successive_args(*expected_yields)
+    end
+
+    it 'parses multiple long flags correctly' do
+      input = ['--help', '--jobs']
+      expected_yields = [
+        ['--help', 0, nil],
+        ['--jobs', 1, nil]
+      ]
+      expect { |b| subject.parse(input, &b) }.to yield_successive_args(*expected_yields)
+    end
+
+    it 'parses a long flag and short flag correctly' do
+      input = ['--help', '-h']
+      expected_yields = [
+        ['--help', 0, nil],
+        ['-h', 1, nil]
+      ]
+      expect { |b| subject.parse(input, &b) }.to yield_successive_args(*expected_yields)
+    end
+
+    it 'parses a short flag when Rex Arguments are in an array correctly' do
+      input = ['-o']
+      expected_yields = [
+        ['-o', 0, nil]
+      ]
+      expect { |b| subject.parse(input, &b) }.to yield_successive_args(*expected_yields)
+    end
+
+    it 'parses a long flag when in arguments array correctly' do
+      input = ['--options']
+      expected_yields = [
+        ['--options', 0, nil]
+      ]
+      expect { |b| subject.parse(input, &b) }.to yield_successive_args(*expected_yields)
+    end
+
+    it 'parses multiple long flags when in an arguments array correctly' do
+      input = ['--quiet', '--options']
+      expected_yields = [
+        ['--quiet', 0, nil],
+        ['--options', 1, nil]
+      ]
+      expect { |b| subject.parse(input, &b) }.to yield_successive_args(*expected_yields)
+    end
+
+    it 'parses a non-existent long flag correctly' do
```

Comment:
I was under the impression that a long argument that does not exist or a typo e.g. `--intreact` (instead of `--interact`) should be dropped and not passed to any other argument that takes in an option.
Hopefully I conveyed my point clearly 😆  I think leaving this up for discussion could be good, but I can also just change it back. 👍